### DISCUSSION TITLE
ModAPI :: roadmap

### DISCUSSION THREAD

**Discussion by radmanplays** [eaglerforge/EaglerForge#6] [Announcements]
- [x] v1.1

added "update" event that fires every client tick

added platform api

added logger api

- [x] v1.1.1

fixed mods not loading after reloading the page

added loading mods from eaglercraftxopts: https://eaglerforge.github.io/guide/Loading-mods-from-eaglercraftxopts.html

added loading mods from the search bar: 
https://eaglerforge.github.io/guide/Loading-mods-from-search-bar.html

- [x] v1.2

updated to u23

added player, gamesettings, blocks, items, materials and enchantments globals

added getFPS() and drawstringwithshadow() and getdisplayHeight() and getdisplayWidth() to mod api

added getUserAgentString() and getGLRenderer() and getGLVersion() to platform api

added a button that lets you launch the game without mods 

- [x] v1.2.1

added empty gui

added unminified offline downloads and web folder

- [x] v1.2.2

fixed build tools 

fixed drawrect

new global: ScaledResolution

added ModAPI.getFONT_HEIGHT()

changed the location of the Mods buttton on the title screen

- [x] v1.2.3

ported eaglerforge to u26.

registered used event "sendchatmessage" more info here : https://eaglerforge.github.io/apidocs/events/addEventListener.html

added the "GlStateManager" global more info here : https://eaglerforge.github.io/apidocs/globals/GlStateManager.html

added a new method to logger, "setlogger" for setting the name of the logger

- [x] v1.2.4

added the network global: `ModAPI.network.(method)`

added a way to send all networking packets : https://eaglerforge.github.io/apidocs/globals/NetworkData.html

added 2 events for only 2 networking packets sent from server (ill add the other packets in later versions)  : https://eaglerforge.github.io/apidocs/events/FromServerEvents.html

added the mods button in the pause menu back(i forgor to add it when i ported EaglerForge to u26)

fixed A LOT of bugs(half of the player global didnt even work)

one more bug still remains until next version : the items and blocks globals dont work

- [x] v1.2.5

fixed the blocks and items globals

- [x] v1.3

removed unminified and made the minified version be able to use `ModAPI.mcinstance`

added `getSpeed` and `isMoving` and `hasMotion` and `getSpeed` and `setSpeed` to the player global

added a new global: "sp" (singleplayer)

added a new global that MUST BE REQUIRED: "server" (for accessing the server ip, etc) 

added 2 new functions to modapi: `ModAPI.reloadchunks()` and `ModAPI.getProfileName()`

added a new event: `frame`

- [ ] v1.3.1

unknown(for now)

**Comment by 0xerorn** [comment]
damn thats crazy

**Comment by Block2222** [comment]
@radmanplays itll be nice to see a current version of this

**Reply by radmanplays** [reply]
done

**Comment by OeildeLynx31** [comment]
@radmanplays Did you abandon this?

**Reply by radmanplays** [reply]
yesnt

**Reply by STPv22** [reply]
I guess that means he is going to deal with it later?
